### ISSUE TITLE
[v.1.11.0] Release Tracker

### ISSUE THREAD

**Issue by atalman** [pytorch/pytorch#72267]
We cut a [release branch](https://github.com/pytorch/pytorch/tree/release/1.11) for the 1.11.0 release.

Our plan from this point from this point is roughly:

* Phase 1 (until 02/17): work on finalizing the release branch
* Phase 2 (after 02/17): perform extended integration/stability/performance testing based on Release Candidate builds.

This issue is for tracking cherry-picks to the release branch.

## Cherry-Pick Criteria

**Phase 1 (until 02/17/22):**

Only low-risk changes may be cherry-picked from master:

1. Fixes to regressions against the most recent minor release (e.g. 1.10 for 1.11 release; see [module: regression issue list](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22module%3A+regression%22+))
2. Critical fixes for: [silent correctness](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+correctness+%28silent%29%22), [backwards compatibility](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+bc-breaking%22+), [crashes](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+crash%22+), [deadlocks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+deadlock%22+), (large) [memory leaks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+memory+usage%22+)
3. Fixes to new features introduced in the most recent minor release (e.g. 1.10.1 for 1.11 release)
4. Test/CI fixes
5. Documentation improvements
6. Compilation fixes or ifdefs required for different versions of the compilers or third-party libraries
7. Release branch specific changes (e.g. change version identifiers)

Any other change requires special dispensation from the release managers (currently @atalman, @seemethere ). If this applies to your change please write "Special Dispensation" in the "Criteria Category:" template below and explain.

**Phase 2 (after 02/17/22):**

Note that changes here require us to rebuild a Release Candidate and restart extended testing (likely delaying the release). Therefore, the only accepted changes are **Release-blocking** critical fixes for: [silent correctness](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+correctness+%28silent%29%22), [backwards compatibility](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+bc-breaking%22+), [crashes](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+crash%22+), [deadlocks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+deadlock%22+), (large) [memory leaks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+memory+usage%22+)

Changes will likely require a discussion with the larger release team over VC or Slack.

## Cherry-Pick Process

1. Ensure your PR has landed in master. This does not apply for release-branch specific changes (see Phase 1 criteria).
2. Create (but do not land) a PR against the [release branch](https://github.com/pytorch/pytorch/tree/release/1.11).
   <details>

    ```bash
    # Find the hash of the commit you want to cherry pick
    # (for example, abcdef12345)
    git log

    git fetch origin release/1.11
    git checkout release/1.11
    git cherry-pick abcdef12345

    # Submit a PR based against 'release/1.10' either:
    # via the GitHub UI
    git push my-fork

    # via the GitHub CLI
    gh pr create --base release/1.11
    ```

    </details>
3. Make a request below with the following format:

```
Link to landed master PR (if applicable):
* 

Link to release branch PR:
* 

Criteria Category:
* 
```

1. Someone from the release team will reply with approved / denied or ask for more information.
2. If approved, someone from the release team will merge your PR once the tests pass. **Do not land the release branch PR yourself.**

**NOTE: Our normal tools (ghstack / ghimport, etc.) do not work on the release branch.**

Please note HUD Link with branch CI status and link to the HUD to be provided here.
[HUD](https://hud2.pytorch.org/hud/pytorch/pytorch/release%2F1.11/0)

### Versions

1.11

**Comment by ejguan** [comment]
Link to landed master PR (if applicable): https://github.com/pytorch/pytorch/pull/71579
Link to release branch PR: https://github.com/pytorch/pytorch/pull/72269
Criteria Category: Critical fixes https://github.com/pytorch/pytorch/issues/1551

**Comment by zou3519** [comment]
Link to landed master PR (if applicable): https://github.com/pytorch/pytorch/pull/72181
Link to release branch PR: https://github.com/pytorch/pytorch/pull/72274
Criteria Category: Critical fixes. See https://github.com/pytorch/functorch/issues/428 for context.

**Comment by zou3519** [comment]
Link to landed master PR (if applicable): n/a
Link to release branch PR: https://github.com/pytorch/pytorch/pull/72275
Criteria Category: Release branch specific changes (removing prototype feature that should not be in release)

**Comment by ejguan** [comment]
Link to landed master PR (if applicable): n/a
Link to release branch PR: https://github.com/pytorch/pytorch/pull/72279
Criteria Category: Release branch specific changes (removing prototype feature that should not be in release)

**Comment by Kyeongpil** [comment]
The mentioned schedules need to be changed as this year is 2022 :)
e.g. 02/17/21 -> 02/17/22

**Comment by ejguan** [comment]
Link to landed master PR (if applicable): https://github.com/pytorch/pytorch/pull/70435 and https://github.com/pytorch/pytorch/pull/72260
Link to release branch PR: https://github.com/pytorch/pytorch/pull/72335
Criteria Category: Fixes to new features introduced in the most recent minor release (And, this PR would unblock the release of a few downstream libraries like TorchData, TorchText)

**Comment by malfet** [comment]
Pushed https://github.com/pytorch/pytorch/commit/ff6c3487625b3f214ea9e169fe1bbb045c07943a to revert minor version update

**Comment by malfet** [comment]
Link to landed main PR:
 * https://github.com/pytorch/pytorch/pull/72290
Link to cherry-pick PR:
  * https://github.com/pytorch/pytorch/pull/72356
 
Criteria category: Fixes build

**Comment by malfet** [comment]
Link to landed trunk PR:
* https://github.com/pytorch/pytorch/pull/72350

Link to cherry-pick PR:
* https://github.com/pytorch/pytorch/pull/72439

Criteria category: Fixes CI regressions

**Comment by malfet** [comment]
Link to landed trunk PR:
* https://github.com/pytorch/pytorch/pull/72433

Link to a cherry-pick PR:
* https://github.com/pytorch/pytorch/pull/72440

Criteria category: Fixes CI regressions

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
https://github.com/pytorch/builder/pull/955

Link to release branch PR:
https://github.com/pytorch/builder/pull/957

Criteria category: Fixes CI regressions

**Comment by IvanYashchuk** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/72357

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/72513

Criteria Category:
* Fixes INTERNAL_ASSERT triggers, see https://github.com/pytorch/pytorch/issues/71645

**Comment by IvanYashchuk** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/72312

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/72514

Criteria Category:
* Documentation improvements: see https://github.com/pytorch/pytorch/pull/68007#discussion_r798973898

**Comment by anjali411** [comment]
Link to landed master PR (if applicable):

n/a
Link to release branch PR:

https://github.com/pytorch/pytorch/pull/72604
Criteria Category:

Release branch specific changes (disable complex32)
XLA side change: https://github.com/pytorch/xla/pull/3340

**Comment by malfet** [comment]
Link to landed master PR:
* https://github.com/pytorch/pytorch/pull/72567

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/72635

Criteria Category: CI bugfixes  (docs are no longer generated for tagged builds)

**Comment by malfet** [comment]
Link to landed master PR:
* https://github.com/pytorch/pytorch/pull/72628

Link to the release branch PR:
* https://github.com/pytorch/pytorch/pull/72652

Criteria Category: CI bugfixes (versions are not set correctly to tagged builds)

**Comment by soulitzer** [comment]
Link to landed master PR:
- https://github.com/pytorch/pytorch/pull/72627

Link to release branch PR:
- https://github.com/pytorch/pytorch/pull/72656

Criteria Category: critical fix for negative python refcount

**Comment by eellison** [comment]
Link to landed master PR:
https://github.com/pytorch/pytorch/pull/72616
Link to release branch PR: 
 https://github.com/pytorch/pytorch/pull/72678 

Criteria: critical fix for dynamic shape fusion crash

**Comment by jbschlosser** [comment]
Link to landed master PR: #72436
Link to release branch PR: #72697
Criteria: Fix for regression

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/72719

Link to release branch PR:
*  https://github.com/pytorch/pytorch/pull/72732

Criteria Category:
*  Critical CI fixes

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
*  NA

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/72739/

Criteria Category:
* Critical CI fixes

**Comment by NivekT** [comment]
Link to landed master PR:
* https://github.com/pytorch/pytorch/pull/72475
* https://github.com/pytorch/pytorch/pull/72476
* https://github.com/pytorch/pytorch/pull/72618

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/72801

Criteria Category:
* Documentation improvements - all changes in these PRs are made to docstrings.

**Comment by ejguan** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/72715

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/72800

Criteria Category:
* Fixes to new features introduced in the most recent minor release (e.g. 1.10.1 for 1.11 release)

**Comment by jbschlosser** [comment]
Master branch PR: #61439
Release branch PR: #72826
Criteria: Special dispensation - important NVIDIA-contributed approximation to GELU

**Comment by malfet** [comment]
Link to landed master PR: 
* https://github.com/pytorch/pytorch/pull/72532

Link to release branch PR: 
* https://github.com/pytorch/pytorch/pull/72734

Criteria: Regression from previous release (ONNX)

---
@malfet: merged

**Comment by bdhirsh** [comment]
Link to landed master PR:
- https://github.com/pytorch/pytorch/pull/72543

Link to release branch PR:
- https://github.com/pytorch/pytorch/pull/72913

**Comment by bdhirsh** [comment]
Link to landed master PR:
- https://github.com/pytorch/pytorch/pull/72540

Link to release branch PR:
- https://github.com/pytorch/pytorch/pull/72914

Criteria: Marking `torch.TypedStorage` API as private before the 1.11 release, since it doesn't have public docs yet.

**Comment by jerryzh168** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/72311

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/72929

Criteria Category:
* Documentation for quantization api

**Comment by eellison** [comment]
https://github.com/pytorch/pytorch/pull/72937 

Linked to landed master PR (if applicable):

https://github.com/pytorch/pytorch/pull/72638 https://github.com/pytorch/pytorch/pull/72639

Link to release branch PR:

https://github.com/pytorch/pytorch/pull/72937

Critical Category:

Documentation for dynamic shape fusion api

**Comment by yeounoh** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/72496

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/72938

Criteria Category:
* Enabling XLA workflow for XLA testing and building for PyTorch/XLA 1.11 release

**Comment by VitalyFedyunin** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/72919 (Landing ATM)

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/72954

Criteria Category:
* Documentation for native_channels_shuffle

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/72806

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/72959

Criteria Category:
* fixes regression in libtorch

**Comment by malfet** [comment]
Link to landed trunk PR:
* https://github.com/pytorch/pytorch/pull/72844

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/73017

Criteria Category:
 - CI regression fixes

**Comment by seemethere** [comment]
Link to landed trunk PR:
* https://github.com/pytorch/pytorch/pull/72903

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/73027

Criteria Category:
 - fix regressions in releng pipelines

**Comment by jbschlosser** [comment]
Link to landed master PR: #73014
Link to release branch PR: #73049
Criteria category: Documentation (fix some formatting regressions)

**Comment by malfet** [comment]
Link to the main trunk PR:
*  https://github.com/pytorch/pytorch/pull/72794

Link to the release branch PR: 
* https://github.com/pytorch/pytorch/pull/73122

Criteria Category: CI Fixes

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/builder/pull/963

Link to release branch PR:
* https://github.com/pytorch/builder/pull/968

Criteria Category:
* CI Fixes

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/73013

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/73312

Criteria Category:
*  Documentation improvement

**Comment by cbalioglu** [comment]
Link to landed master PR:
- #71746

Link to release branch PR:
- #73357

Criteria Category:
- Fixes external customer regression

**Comment by cbalioglu** [comment]
Link to landed master PRs:
- #73144
- #73166
- #73167
- #73149

Link to release branch PR:
- #73358

Criteria Category:
- Fixes external customer regressions
  - #73114
  -  #73121

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/builder/pull/971

Link to release branch PR:
* https://github.com/pytorch/builder/pull/975

Criteria Category:
* Critical CI fixes

**Comment by malfet** [comment]
Link to landed trunk PR:
* https://github.com/pytorch/pytorch/pull/73125

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/73365

Criterial Category:
* Documentation updates

**Comment by malfet** [comment]
Cherry-picked https://github.com/pytorch/pytorch/commit/9647fb7d1823099efc0c56cb9adc9524e14951e6 to fix libtorch builds

**Comment by malfet** [comment]
Follow up after discussion with @seemethere and @atalman :
CUDA+OS release matrix for 1.11:
 - Linux: 10.2, 11.1 (deprecated), 11.3 (default), 11.5 (exists in `nvidia` channel, depends on conda-forge)
 - Windows: 11.3 (cudatoolkit exists in `default`) channel), 11.5 (exists in `nvidia` channel)

**Comment by lw** [comment]
Commit 055052bf64ab9ed6196324df38c03107c574f1fe (landed yesterday) introduced a regression which makes it impossible to build c10d extensions, because it adds a `c10d/debug.h` header but it doesn't install it (but which is recursively included from `c10d/ProcessGroup.h`). [This](https://app.circleci.com/pipelines/github/facebookresearch/fairring/343/workflows/24b8e6f8-b5f2-41d2-b428-b054e2c100c8/jobs/4416) is the failure I'm getting in my CI. It is imperative to fix this before the v1.11 stable release. I'm surprised this passed CI: I thought we had a test that builds a simple c10d extension?

CC @cbalioglu

**Comment by cbalioglu** [comment]
@lw the original PR is #73144. As you can see both internal and external CIs were green. Do you know where that test is defined?

**Comment by cbalioglu** [comment]
@lw the original PR is #73144. As you can see both internal and external CIs were green. Do you know where that test is defined?

**Comment by lw** [comment]
I opened a separate issue to avoid polluting this thread: https://github.com/pytorch/pytorch/issues/73421

**Comment by lw** [comment]
Link to landed trunk PR:
https://github.com/pytorch/pytorch/pull/73422

Link to release branch PR:
https://github.com/pytorch/pytorch/pull/73497

Criterial Category:
Backwards compatibility (build failure for 3rd party extensions)

This fixes a regression introduced into the release branch just a few days ago, after it was cut. It doesn't cause functional changes (it doesn't touch Python or C++ code) hence no additional testing of a new RC is required. It's just two straightforward lines of code.

**Comment by lw** [comment]
@malfet @seemethere Could you take a look at https://github.com/pytorch/pytorch/issues/72267#issuecomment-1054063875? Thanks!

**Comment by atalman** [comment]
Please note: We don't accept cherry-picks anymore on this Issue. Thank you!

**Comment by cbalioglu** [comment]
> Please note: We don't accept cherry-picks anymore on this Issue. Thank you!

@atalman is there going to be another time window for cherry-picks? @lw's work (a fairly small PR) is a follow-up to #73358 and fixes an important regression.

**Comment by malfet** [comment]
@cbalioglu abovementioned PR was merged, but it will probably be a good indication that something as non-trivial as https://github.com/pytorch/pytorch/pull/73358 should not have been picked into the branch one day before final RC

**Comment by eellison** [comment]
https://github.com/pytorch/pytorch/pull/73802 - data race issue for multi-threaded inference with dynamic shapes on GPU as seen in https://github.com/pytorch/pytorch/issues/73789. sorry I know this is late :/

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/73805

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/73900

Criteria Category:
* Critical Fix to CI

**Comment by atalman** [comment]
Closing Issue since no more Cherry Pickings are accepted.
